src/calculate.py

```diff
@@ -77,33 +95,70 @@ def compute_monthly(self, month):
                 if reading.min_temp:
                     total_min_temp += reading.min_temp
                 if reading.humidity:
-
                     total_humidity += reading.humidity
                 count += 1
 
+        if count == 0:
+            result = {"avg_max_temp": 0, "avg_min_temp": 0, "avg_humidity": 0}
+        else:
+            result = {
+                "avg_max_temp": f"Highest Average {total_max_temp // count}C",
+                "avg_min_temp": f"Lowest AVerage {total_min_temp // count}C",
+                "avg_humidity": f"Average Humidity {total_humidity // count}%",
+            }
 
-        return MonthlyResult(
-            total_max_temp // count, total_min_temp // count, total_humidity // count
-        )
+        return result
+
+    def print_monthly_report(self, result):
+        """
+        Description: print the dict returned from compute_monthly
+
+        Args:
+            result
+        """
+        [print(value) for value in result.values()]
+
+
+class ChartDataGeneration:
+    def __init__(self, readings: List[WeatherReading]):
+        self.readings = readings
 
     """
-    Generates  chart
-    """
-    def generate_chart_data(self, month):
+    Description: Generates a dictionary of  data for a specified month.
 
-        chart_data = defaultdict(list)
+    Argument: month
 
+    Returns:
+        Dict[str, List[int]]: A dictionary where the keys are days of the month and the values are lists containing the maximum and minimum temperatures recorded on that day.
+    """
 
-        if month:
-            year=month.split("/")[0]
-            m=month.split("/")[1]
-            year_month_combination=year+'-'+m
+    def generate_chart_data(self, month: str) -> Dict[str, List[int]]:
+        chart_data = defaultdict(list)
 
+        year, month_ = month.split("/")
+        year_month_combination = f"{year}-{month_}"
 
         for reading in self.readings:
             if reading.date.startswith(year_month_combination):
                 day = reading.date.split("-")[-1]
                 chart_data[day].append(reading.max_temp)
                 chart_data[day].append(reading.min_temp)
-        logging.info("done with chart data")
-        return ChartData(chart_data)
+        # self.print_non_none_days(chart_data)
+        return chart_data
+
+    """
+    Description: Prints the days of the month with recorded temperature data.
+
+    Args:
+        chart_data (Dict[str, List[int]])
+
+    Returns:
+        None
+    """
+
+    def print_non_none_days(self, chart_data: Dict[str, List[int]]):
+        for day, temps in chart_data.items():
+            max_temp, min_temp = temps
+            if max_temp is not None and min_temp is not None:
```

Comment:
still unsresolved

Reply:
Now its getting printed in the main, not in this class.